src/document.jl

```diff
@@ -90,6 +90,13 @@ function parse_file(filename::AbstractString)
     XMLDocument(p)
 end
 
+function parse_file(filename::AbstractString, encoding, options::Integer)
+    p = ccall((:xmlReadFile,libxml2), Xptr, (Cstring, Ptr{Cchar}, Cint),
```

Comment:
The problem with not using `Cstring` is that you get silent truncation of `filename` strings with embedded NUL chars.